a00_qpip/plugin.py

```diff
@@ -170,6 +175,8 @@ def check_deps_and_prompt_install(self, additional_plugins=[], force_gui=False):
             dialog = MainDialog(
                 libs.values(), self._check_on_startup(), self._check_on_install()
             )
+            if self.test_mode:
+                return dialog
```

Comment:
If we split this in two methods:

- `check_deps` or `get_dependency_dialog`
- `prompt_install`

(or similar), we could test `get_dependency_dialog` while we leave the `prompt_install` and leave `prompt_install`, this way we could get rid of `test_mode` completely.